cookbooks/mediawiki/templates/default/LocalSettings.php.erb

```diff
@@ -98,6 +98,23 @@ $wgFileExtensions[] = 'osm';
 	$wgFileExtensions[] = '<%= mw_extra_file_extension %>';
 <% end -%>
 
+# Add OSM XML file format per http://www.iana.org/assignments/media-types/media-types.xhtml
+# Shout out to Paul Norman for reserving this.
+$wgXMLMimeTypes[] = 'osm' => 'application/vnd.openstreetmap.data+xml';
+
+# Helps MimeMagic determine XML-based formats and chooses the correct MimeType
+# for .osm files.
+$wgXMLMimeTypes = array(
+	'http://www.w3.org/2000/svg:svg' => 'image/svg+xml',
+	'svg' => 'image/svg+xml',
+	'http://www.lysator.liu.se/~alla/dia/:diagram' => 'application/x-dia-diagram',
+	'http://www.w3.org/1999/xhtml:html' => 'text/html',
+	'html' => 'text/html',
+	'osm' => 'application/vnd.openstreetmap.data+xml' );
+
+$wgTrustedMediaFormats[] = 'application/xml'
+$wgTrustedMediaFormats[] = 'application/vnd.openstreetmap.data+xml'
+
```

Comment:
Why do we need to add `application/xml` as well as the OpenStreetMap one?

Reply:
In theory, the second config value should work perfectly well, but I can't test it to verify. This whitelists both formats in case it doesn't work as expected. (covering bases) I'll add only one to reduce the cruft.